### ISSUE TITLE
Using a combined path in python intermediate array fetching no longer works

### ISSUE THREAD

**Issue by svniemeijer** [stcorp/coda#91]
When using an AEOLUS L2B file, this works:

```
>>> coda.fetch(product, 'mie_geolocation', -1,'windresult_geolocation', 'latitude_cog')
```

But this doesn't:

```
>>> coda.fetch(product, 'mie_geolocation', -1,'windresult_geolocation/latitude_cog')
```

```
coda.CodacError: coda_type_get_record_field_index_from_name(): record does not contain a field named 'windresult_geolocation/latitude_cog'
```

We should be using a `cursor_goto` to traverse any string arguments.

**Comment by srepmub** [comment]
oh darnit. it looks like I mistakenly had assumed that path 'elements' are always field names. this means the 'reldepth' and goto_parent strategy optimization also doesn't work. for now I reverted the type_path and goto_parent optimizations..

if we want such an optimization again, we could perhaps detect the case that we only have field lookups (and integers) in the whole 'path'. in this case we can cache field indices and backtrack without having to call get_cursor_depth (twice).. can we easily see that a node expression is just a field lookup? (alternatively we could also check that the whole node expression matches a field name in the current record).

**Comment by srepmub** [comment]
(for just field names joined with '/' we could also cache field indices of course..)

**Comment by svniemeijer** [comment]
An alternative to using goto_parent is to create a copy of the cursor and using that to navigate the subtree:

```
import copy
cursor = coda.Cursor()
cursor2 = copy.deepcopy(cursor)
```

This copied cursor can then be thrown away instead of having to call goto_parent.

Not sure how this works with the type_path part.

**Comment by srepmub** [comment]
so this would probably be quite slow..

**Comment by svniemeijer** [comment]
This was fixed in version 2.24.2
